### PR TITLE
[Dialog]: Update documentation and tests for nested dialog

### PR DESCRIPTION
**Pull request by xanxan**
https://funidata.atlassian.net/browse/DS-359

- Storybook documentation, unit tests and e2e tests updated

### CODE REVIEW

File: test/visual-regression/dialog.spec.ts-snapshots/nested-dialog-berry-2-Mobile-Chrome-Big-Landscape-linux.png
Comment (MayaMarjut) [inline test/visual-regression/dialog.spec.ts-snapshots/nested-dialog-berry-2-Mobile-Chrome-Big-Landscape-linux.png:1]: Focus style is being cropped on this screenwidth.

File: test/visual-regression/dialog.spec.ts-snapshots/nested-dialog-berry-1-Mobile-Safari-linux.png
Comment (MayaMarjut) [inline test/visual-regression/dialog.spec.ts-snapshots/nested-dialog-berry-1-Mobile-Safari-linux.png:1]: We should ask input from designers how they want these buttons to be stacked on small screen

**Comment by MayaMarjut** [comment]
Currently if screenreader user closes a second dialog from 'x'-button, keyboard is focusing on 'Save and open next dialog' button - without announcing the name of a last opened dialog. This can be confusing. Better functionality would be moving the focus to the header section of the newly opened dialog, as it does now when the user simply moves to next dialog by pressing "Save and open next dialog" button. This same happens when user closes the dialog with 'ESC'.
Screenshot 2024-09-13 at 7 53 46

File: test/visual-regression/dialog.spec.ts-snapshots/nested-dialog-berry-2-Mobile-Chrome-Big-Landscape-linux.png
Comment (RiinaKuu) [reply test/visual-regression/dialog.spec.ts-snapshots/nested-dialog-berry-2-Mobile-Chrome-Big-Landscape-linux.png:1]: This was due to the fact that the whole dialog content was not in the screenshot and view has not been scrolled. Added scrolling to these tests and now whole focus is visible.

**Comment by RiinaKuu** [comment]
> Currently if screenreader user closes a second dialog from 'x'-button, keyboard is focusing on 'Save and open next dialog' button - without announcing the name of a last opened dialog. This can be confusing. Better functionality would be moving the focus to the header section of the newly opened dialog, as it does now when the user simply moves to next dialog by pressing "Save and open next dialog" button. This same happens when user closes the dialog with 'ESC'.

It is part of Dialog's closing logic to move focus where the dialog initially was opened. Your suggestion could be an improvement to think about later on if we want to change the logic for screen reader users.

**Review by RiinaKuu — APPROVED** [review]

**Pull request merged**